testserver/testserver.go

```diff
@@ -263,7 +264,7 @@ func (ts *TestServer) pollListeningURLFile() error {
 		time.Sleep(100 * time.Millisecond)
 	}
 
-	u, err := url.Parse(string(data))
+	u, err := url.Parse(string(bytes.TrimSpace(data)))
```

Comment:
In this case we're just seeing the trailing newline in the url file. If you can help me conceive of a test that would exercise legitimate spaces in the URL I'd be happy to adding urlencoding.